lib/ecto_trail/ecto_trail.ex

```diff
@@ -111,6 +124,35 @@ defmodule EctoTrail do
     |> run_logging_transaction(repo, changeset, actor_id)
   end
 
+  @doc """
+  Call `c:Ecto.Repo.update/2` operation for every changeset and store all changes in a `change_log` table.
+
+  Return and options are similar to `c:Ecto.Repo.update_all/3`.
+  """
+  @spec update_all_and_log(changesets :: list(Ecto.Changeset.t),
+                        actor_id :: String.T,
+                        opts :: Keyword.t) ::
+        {non_neg_integer, nil} |
+        {:error, Ecto.Changeset.t}
+  def update_all_and_log(repo, changesets, actor_id, opts \\ []) do
```

Comment:
Functions should have a @spec type specification.

Reply:
fixed